synapse/state/__init__.py

```diff
@@ -422,66 +422,57 @@ async def compute_event_context(
 
     async def compute_event_context_for_batched(
         self,
-        event: EventBase,
-        state_ids_before_event: StateMap[str],
-        current_state_group: int,
-    ) -> EventContext:
+        events_and_context: List[Tuple[EventBase, EventContext]],
+        prev_group: int,
+        state_ids_before_event: StateMap,
+    ) -> List[Tuple[EventBase, EventContext]]:
         """
         Generate an event context for an event that has not yet been persisted to the
         database. Intended for use with events that are created to be persisted in a batch.
         Args:
-            event: the event the context is being computed for
-            state_ids_before_event: a state map consisting of the state ids of the events
-            created prior to this event.
-            current_state_group: the current state group before the event.
+            events_and_context: a list of events and their associated contexts
+            prev_group: the state group of the last event persisted before the batched events
+            were created
+            state_ids_before_event: a state map consisting of current state ids
```

Comment:
Can we
* clarify that `prev_group` is the state group _after_ the last event
* and rename `state_ids_before_event` to `prev_state_ids`? By naming both parameters with the same `prev_` prefix, we can convey that they're closely related.

```suggestion
            prev_group: the state group after the last event persisted before the
                batched events were created
            prev_state_ids: a state map consisting of current state ids before the
                batched events
```